Core/Utilities/ElasticSearchDB.py

```diff
@@ -364,10 +373,22 @@ def getUniqueValue( self, indexName, key, orderBy = False ):
     return S_OK( values )
 
 
-def generateFullIndexName( indexName ):
+def generateFullIndexName( indexName, period = 'day' ):
   """
   Given an index prefix we create the actual index name. Each day an index is created.
   :param str indexName: it is the name of the index
+  :param str period: We can specify, which kind of indexes will be created. Currently only daily and monthly indexes are supported.
   """
+    
   today = datetime.today().strftime( "%Y-%m-%d" )
-  return "%s-%s" % ( indexName, today )
+  index = ''
+  if period.lower() not in ['day', 'month']:  # if the period is not correct, we use daily indexes.
```

Comment:
I would say it will be safer. At the moment we are sure that we do not call period=None. But as soon as we add a new Monitoring type, this can happen.